File: app/src/main/java/com/example/rossinyamaral/bank/statementsScreen/StatementsConfigurator.java
package com.example.rossinyamaral.bank.statementsScreen;

import java.lang.ref.WeakReference;


public enum StatementsConfigurator {
    INSTANCE;

    public void configure(StatementsActivity activity) {

        StatementsRouter router = new StatementsRouter();
        router.activity = new WeakReference<>(activity);

        StatementsPresenter presenter = new StatementsPresenter();
        presenter.output = new WeakReference<StatementsActivityInput>(activity);

        StatementsInteractor interactor = new StatementsInteractor();
        interactor.output = presenter;

        if (activity.output == null) {
            activity.output = interactor;
        }
        if (activity.router == null) {
            activity.router = router;
        }
    }
}

File: SantanderApp/app/src/main/java/com/qintess/santanderapp/ui/view/statementsScreen/StatementsConfigurator.kt
package com.qintess.santanderapp.ui.view.statementsScreen

import java.lang.ref.WeakReference

enum class StatementsConfigurator {
    INSTANCE;

    fun configure(activity: StatementsActivity) {
        val presenter = StatementsPresenter()
        presenter.output = WeakReference(activity)

        val interactor = StatementsInteractor()
        interactor.output = presenter

        if (activity.output == null) {
            activity.output = interactor
        }
    }
}